docs/tools/config_system.md

```diff
@@ -1,23 +1,135 @@
+events.shared-highprio-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_HIGHPRIO_STACKSIZE")
+events.shared-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_STACKSIZE")
+<output truncated for brevity>
+```
+
+You may find that configuration naming does not always describe the parameter's purpose adequately. Include the `-v` switch to include the help text defined with the configuration parameter, where the value of the configuration parameter is defined, and other details. The command `mbed compile --config -v` in the same application as above prints:
+```
+Configuration parameters
+------------------------
+Name: cellular.random_max_start_delay
+    Description: Maximum random delay value used in start-up sequence in milliseconds
+    Defined by: library:cellular
+    No value set
+Name: cellular.use-apn-lookup
+    Description: Use APN database lookup
+    Defined by: library:cellular
+    Macro name: MBED_CONF_CELLULAR_USE_APN_LOOKUP
+    Value: 1 (set by library:cellular)
+Name: configuration-store.storage_disable
+    Description: Configuration parameter to disable flash storage if present. Default = 0, implying that by default flash storage is used if present.
+    Defined by: library:configuration-store
+    No value set
+<output truncated for brevity>
+```
+
+### Using configuration data in code
+
+When compiling or exporting, the configuration system generates C preprocessor macro definitions of the configuration parameters. The configuration system writes these definitions in a file named `mbed_config.h` located in the build directory. When compiling the same example as the prior section for target `K64F`, the `mbed_config.h` file includes this snippet (note that the order of the definitions may be different):
+
+```C
+// Automatically generated configuration file.
+// DO NOT EDIT, content will be overwritten.
+
+#ifndef __MBED_CONFIG_DATA__
+#define __MBED_CONFIG_DATA__
+
+// Configuration parameters
+#define MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY         0 // set by library:cellular
+#define MBED_CONF_CELLULAR_USE_APN_LOOKUP                 1 // set by library:cellular
+<file truncated for brevity>
+```
+
+<span class="notes">**Note:** A macro definition will not be generated for a parameter that doesn't have a value.</span>
+
+
+The names of the macro for a configuration parameter is either a prefixed name or explicitly specified by `macro_name`. A prefixed name is constructed from the prefix `MBED_CONF_`, followed by the name of the library or `APP`, followed by the name of the parameter. The prefixed name is then capitalized and converted to a valid C macro name. For example, the `random_may_start_delay` configuration parameter in the library `cellular` is converted to `MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY`.
+
+The Mbed OS build tools instruct the compiler to process the file `mbed_config.h` as if it were the first include of any C or C++ source file, so you do not have to include `mbed_config.h` manually.
```

Comment:
Via what mechanism? (I'm not sure if it's best to add that info here, but I'm curious.)

Reply:
That's `-include` (should be `-imacros`) for gcc, `-preinclude` for both ArmC5 and IAR EWARM. I don't think that's really relevant, as you mentioned. I also reserve the right to change that implementation detail.